include/cinder/Log.h

```diff
@@ -159,92 +196,49 @@ class LoggerSystem : public Logger {
 #endif
 };
 
-//! \brief Logger that can log to multiple other Loggers.
+//! \brief LogManager manages a stack of all active Loggers.
 //!
-//! This is primarily used by LogManager as it's base Logger, when multiple log outputs are enabled (ex. console and file)
-class LoggerMulti : public Logger {
-  public:
-	void add( Logger *logger )							{ mLoggers.push_back( std::unique_ptr<Logger>( logger ) ); }
-	void add( std::unique_ptr<Logger> &&logger )		{ mLoggers.emplace_back( move( logger ) ); }
-
-	template <typename LoggerT>
-	LoggerT* findType();
-
-	void remove( Logger *logger );
-
-	const std::vector<std::unique_ptr<Logger> >& getLoggers() const	{ return mLoggers; }
-
-	void write( const Metadata &meta, const std::string &text ) override;
-
-  private:
-	std::vector<std::unique_ptr<Logger> >	mLoggers;
-};
-
+//! LogManager's default state contains a single LoggerConsole.  LogManager allows for adding and removing Loggers via their pointer values.
 class LogManager {
 public:
 	// Returns a pointer to the shared instance. To enable logging during shutdown, this instance is leaked at shutdown.
 	static LogManager* instance()	{ return sInstance; }
 	//! Destroys the shared instance. Useful to remove false positives with leak detectors like valgrind.
 	static void destroyInstance()	{ delete sInstance; }
-	//! Restores LogManager to its default state.
+	//! Restores LogManager to its default state - a single LoggerConsole.
 	void restoreToDefault();
 
-	//! Resets the current Logger stack so only \a logger exists.
-	void resetLogger( Logger *logger );
+	//! Removes all loggers from the stack.
+	void clearLoggers();
+	//! Resets the current Logger stack so only \p logger exists.
+	void resetLogger( const LoggerRef& logger );
 	//! Adds \a logger to the current stack of loggers.
-	void addLogger( Logger *logger );
+	void addLogger( const LoggerRef& logger );
 	//! Remove \a logger to the current stack of loggers.
-	void removeLogger( Logger *logger );
-	//! Returns a pointer to the current base Logger instance.
-	Logger* getLogger()	{ return mLogger.get(); }
-	//! Returns a logger of a specifc type, or nullptr is that type of Logger is currently not in use.
+	void removeLogger( const LoggerRef& logger );
+	//! Returns a vector of raw pointers that contain all active loggers of a specifc type.
 	template<typename LoggerT>
-	LoggerT* getLogger();
-	//! Returns a vector of all current loggers
-	std::vector<Logger *> getAllLoggers();
-	//! Returns the mutex used for thread safe loggers. Also used when adding or resetting new loggers.
+	std::vector<LoggerT*> getLoggers();
```

Comment:
Despite my suggestion above to return LoggerRef (wasn't thinking to deeply), what if we did it [like this](https://github.com/richardeakin/Cinder/commit/3e5d810fa29a5cfa7e4c03473ab7e3b380967d6e)?  That way if the type of logger you're requesting has special parameters you have the right type in the vector and can act on them directly, without needing to type cast again.

If this is agreeable, then I'll go ahead and merge the PR along with that change.